x-pack/filebeat/input/http_endpoint/input_test.go

```diff
@@ -55,6 +56,60 @@ var serverPoolTests = []struct {
 			{"json": mapstr.M{"c": int64(3)}},
 		},
 	},
+	{
+		name:   "put",
+		method: http.MethodPut,
+		cfgs: []*httpEndpoint{{
+			addr: "127.0.0.1:9001",
+			config: config{
+				Method:        http.MethodPut,
+				ResponseCode:  200,
+				ResponseBody:  `{"message": "success"}`,
+				ListenAddress: "127.0.0.1",
+				ListenPort:    "9001",
+				URL:           "/",
+				Prefix:        "json",
+				ContentType:   "application/json",
+			},
+		}},
+		events: []target{
+			{url: "http://127.0.0.1:9001/", event: `{"a":1}`},
+			{url: "http://127.0.0.1:9001/", event: `{"b":2}`},
+			{url: "http://127.0.0.1:9001/", event: `{"c":3}`},
```

Comment:
How does it behave if there is no request body for a `PUT`

Reply:
Exactly the same way that a zero body POST would behave.